src/wms/src/main/java/org/geoserver/wms/MapLayerInfo.java

```diff
@@ -337,6 +355,43 @@ public int hashCode() {
         return result;
     }
 
+    private boolean userMapCRSForWFSNG(
+            FeatureTypeInfo resource, CoordinateReferenceSystem coordinateReferenceSystem)
+            throws IOException {
+        // verify the resource is WFS-NG and contains Other SRS in feature metadata
+        if (resource.getStore().getConnectionParameters().get(WFSDataStoreFactory.USEDEFAULTSRS.key)
+                        == null
+                || resource.getMetadata().get(FeatureTypeInfo.OTHER_SRS) == null) return false;
+        // do nothing if datastore is configure to stay with native remote srs
+        if (Boolean.valueOf(
+                resource.getStore()
+                        .getConnectionParameters()
+                        .get(WFSDataStoreFactory.USEDEFAULTSRS.key)
+                        .toString())) return false;
+
+        // check if map crs is part of other srs, if yes send put it sindie hend
+        // read all identifiers of this CRS into a list
+        List<String> identifiers =
+                coordinateReferenceSystem
+                        .getIdentifiers()
+                        .stream()
+                        .map(r -> r.toString())
+                        .collect(Collectors.toList());
+        String otherSRSList = (String) resource.getMetadata().get(FeatureTypeInfo.OTHER_SRS);
+        // check if mapCRS is supported in remote wfs layer
+        for (String crs : identifiers)
+            if (otherSRSList.contains(crs)) {
+                // also verify axis order if matched
+                try {
+                    return CRS.equalsIgnoreMetadata(CRS.decode(crs), coordinateReferenceSystem);
+                } catch (Exception e) {
+                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                }
+            }
+
+        return false;
+    }
+
```

Comment:
the method that during a GetMap lookups the requested CRS inside FeatureTypeInfo metadata to skip re-projection. Method also verified that the datastore is WFS-NG configured with`usedefaultsrs` set to false.